### PR TITLE
lnrpc+rpcserver: display payreq via listpayments

### PR DESCRIPTION
**Pull request by cfromknecht**
blocked by #2761

### CODE REVIEW

**Comment by halseth** [comment]
Conflicts with #2761

**Comment by cfromknecht** [comment]
@halseth do you want to merge this #2761? first and last commit could probably be cherry-picked, you'd probably know best about where to store it

**Comment by halseth** [comment]
I'll make sure #2761 adds the `payreq` field such that this can easily be rebased on top :)

**Comment by Roasbeef** [comment]
Needs a rebase!

**Comment by cfromknecht** [comment]
@roasbeef @halseth PTAL

File: lnd_test.go [hunk 1]
```diff
@@ -3825,12 +3825,19 @@ func testListPayments(net *lntest.NetworkHarness, t *harnessTest) {
...
-	// been no fee applied.
+	// As we made a single-hop direct payment, there should have been no fee
+	// applied.
 	if p.Fee != 0 {
 		t.Fatalf("incorrect Fee, got %v, want %v", p.Fee, 0)
 	}
 
+	// Finally, verify that the payment request returned by the rpc matches
+	// the invoice that we paid.
+	if !reflect.DeepEqual(p.PaymentRequest, invoiceResp.PaymentRequest) {
```
Comment (wpaulino) [inline lnd_test.go]: Non-blocking nit: could've used a simple equality check.

Comment (cfromknecht) [reply lnd_test.go]: fixed

File: rpcserver.go [hunk 2]
```diff
@@ -4149,6 +4152,7 @@ func (r *rpcServer) ListPayments(ctx context.Context,
 			Path:            path,
 			Fee:             int64(route.TotalFees().ToSatoshis()),
 			PaymentPreimage: hex.EncodeToString(preimage[:]),
+			PaymentRequest:  string(payment.Info.PaymentRequest),
```
Comment (halseth) [inline rpcserver.go]: I think this is okay, but just to be sure: does this work if the `PaymentRequest` is nil?

File: lntest/itest/lnd_test.go [hunk 3]
```diff
@@ -3826,12 +3826,19 @@ func testListPayments(net *lntest.NetworkHarness, t *harnessTest) {
...
-	// been no fee applied.
+	// As we made a single-hop direct payment, there should have been no fee
+	// applied.
 	if p.Fee != 0 {
 		t.Fatalf("incorrect Fee, got %v, want %v", p.Fee, 0)
 	}
 
+	// Finally, verify that the payment request returned by the rpc matches
+	// the invoice that we paid.
+	if p.PaymentRequest != invoiceResp.PaymentRequest {
```
Comment (halseth) [inline lntest/itest/lnd_test.go]: also add a check for a payment with no payment request?

File: rpcserver.go (hunk 2 above)
Comment (cfromknecht) [reply rpcserver.go]: looks okay: https://play.golang.org/p/N3ss9fZDYpr

File: lntest/itest/lnd_test.go (hunk 3 above)
Comment (cfromknecht) [reply lntest/itest/lnd_test.go]: added

**Comment by cfromknecht** [comment]
@halseth ptal

**Pull request merged**